### ISSUE TITLE
Renaming 'uuid' Attributes

### ISSUE THREAD

**Issue by pfrybar** [ModelSEED/ModelSEED#121]
With the new subobject linking/alias system, we're not using UUIDs anymore. We should rename the "uuid" attribute to reflect this. Originally I was thinking just "id", but there are quite a few objects with their own "id" attributes already. We could rename these, or maybe use something different. "md5" perhaps?

**Comment by cshenry** [comment]
You know, I don't want to call it "md5" because that's just switching from one unique ID to another.
I suggest:

1.) "uid" (afterall, it still is a universal ID)
2.) "_id" (fits with how things are done in other databases... although it clashes with some built in mongodb functionality)
3.) "dbid" or "_dbid"

Personally, I'm leaning toward "dbid" or "uid".

**Comment by devoid** [comment]
For clarification, this is the "non-public" id? E.g. the one that we set within the database layer? And what are we calling the "public" ids. E.g. the "biochemistry/:x/compound/42"?

**Comment by pfrybar** [comment]
Yes, the non-public ids. The public ids (or links) are currently identified by _subobject__uuid (e.g. compound_uuid). Since they are links now instead of uuids, we could change this to _subobject__link (compound_link). I think this is pretty descriptive.

Also for the links/public ids, we should use the format "biochemistry/compounds/42" and not store the provenance id/alias. This allows two reactions to have the same checksum even if their biochemistries have diverged slightly, making the fanned out database more efficient. Also this is smaller so the links will take up less space.
